operator/pkg/resources/configuration/expander.go

```diff
@@ -0,0 +1,105 @@
+package configuration
+
+import (
+	"fmt"
+	"os"
+	"sort"
+	"strings"
+
+	"github.com/redpanda-data/common-go/rpadmin"
+	corev1 "k8s.io/api/core/v1"
+
+	vectorizedv1alpha1 "github.com/redpanda-data/redpanda-operator/operator/api/vectorized/v1alpha1"
+)
+
+// ExpandForBootstrap passes over the input ClusterConfiguration and returns one with
+// all ConfigMapKeyRef and SecretKeyRef entries turned into environment references,
+// together with a list of env var definitions required to satisfy those.
+func ExpandForBootstrap(cfg map[string]vectorizedv1alpha1.ClusterConfigValue) (map[string]vectorizedv1alpha1.ClusterConfigValue, []corev1.EnvVar, error) {
```

Comment:
I think having a distinction between the CRD API and the intermediate template/expandable type would be useful. `ConfigMapKeyRef` and `SecretKeyRef` aren't valid in the template expansion case and `EnvVarRef` isn't really valid in the CRD API case.

Reply:
Yeah, done.